HelloWorld/Program.cs

```diff
@@ -1,2 +1,6 @@
 ﻿//Test Programm
-Console.WriteLine("Hello World");
\ No newline at end of file
+string name;
+Console.WriteLine("Hello World");
+Console.Write("Benutzernamen eingeben: ");
+name = Console.ReadLine();
```

Comment:
Du kannst die Variable auch erst hier definieren mit ` var name = Console.ReadLine();`

"Das Prinzip, eine Variable erst kurz vor ihrer Verwendung zu definieren, wird als "späte Initialisierung" oder "late initialization" bezeichnet. Dies ist eine bewährte Methode in der Programmierung, um den Gültigkeitsbereich einer Variable so klein wie möglich zu halten und die Übersichtlichkeit sowie Wartbarkeit des Codes zu verbessern.

Durch späte Initialisierung wird sichergestellt, dass die Variable nur dann existiert, wenn sie wirklich benötigt wird, was das Risiko von Fehlern, wie ungenutzten oder falsch initialisierten Variablen, verringert."

Durch das var ermittelt der Compiler den Typ der Variable - das ist immer dann sinnvoll, wenn dieser bspw. durch ein Funktionsergebnis festgelegt wird.